### PR TITLE
Add a termination criteria for steady state temperature

### PR DESCRIPTION
**Pull request by gassmoeller**
Essentially the same as the `steady state velocity` termination criterion, just for temperature (e.g. for models without velocity, that only model the transient heat equation).

### CODE REVIEW

File: include/aspect/termination_criteria/steady_temperature.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace TerminationCriteria
+  {
+
+    /**
+     * A class that implements a termination criterion based on steady state
+     * of the root mean square of the velocity field.
```
Comment (tjhei) [inline include/aspect/termination_criteria/steady_temperature.h]: temperature

File: source/termination_criteria/steady_temperature.cc [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+namespace aspect
+{
+  namespace TerminationCriteria
+  {
+    template <int dim>
+    bool
+    SteadyTemperature<dim>::execute(void)
+    {
+      const QGauss<dim> quadrature_formula (this->get_fe()
+                                            .base_element(this->introspection().base_elements.velocities).degree+1);
```
Comment (tjhei) [inline source/termination_criteria/steady_temperature.cc]: why velocity?

**Comment by gassmoeller** [comment]
I addressed the comments and extracted a piece of complicated logic into a separate function to add some unit tests.

**Comment by gassmoeller** [comment]
Is this still good to go?

**Comment by gassmoeller** [comment]
I assume this is good to go, based on Wolfgang's and Timo's approval before.

**Pull request merged**